### ISSUE TITLE
Severe overfitting 

### ISSUE THREAD

**Issue by Ostnie** [titu1994/LSTM-FCN#6]
![image](https://user-images.githubusercontent.com/31840090/40770518-736d7b94-64ed-11e8-804b-59290a8913ed.png)
I don't know the cause of the problem,maybe the data ? I use your model to train my own data., and my X_train size is (8800,128,2), it's a signal data,and the label is (8800,11) ,it's a classfication problem .I change your code like this:
![image](https://user-images.githubusercontent.com/31840090/40771199-4821452c-64ef-11e8-931b-75833861679d.png)

if __name__ == "__main__":
    model = generate_model()
    from keras.callbacks import EarlyStopping, ModelCheckpoint

    early_stopping = EarlyStopping(patience=20)
    checkpointer = ModelCheckpoint('vgg19model_best.h5', verbose=1, save_best_only=True)

    model.compile(loss='categorical_crossentropy',
                  optimizer=Adam(lr=0.001),

                  metrics=['accuracy'])

    print('Train...')
    model.fit(x_train, train_label,
              batch_size=batch_size,
              nb_epoch=500,
              callbacks=[early_stopping, checkpointer],
              validation_data=(x_test, y_test))
    score, acc = model.evaluate(x_test, y_test,
                                batch_size=batch_size) 

It can run directly,but the result is the first picture. Could you please give me some advice?

**Comment by titu1994** [comment]
This codebase takes data in the shape (number of samples, number of variables, timesteps) as it's initial input. Are you following that sequence ?

In that case, you have 128 parameters and 2 timesteps. I suggest you use the Multivariate version of our code since that is designed for such datasets. You can find the link in this repo itself. 

Also, with 8000 data points, you should let it train for at least 100 epochs before concluding that it is overfitting.

**Comment by titu1994** [comment]
As to the models hyperparameters, you have chosen lstm of 128 cells. That may be too large and therefore cause overfitting as well. Try 8 or 64.

**Comment by Ostnie** [comment]
@titu1994 Yes, my data is in the shape(batchsize,features,timesteps),and I change it to (2,128),and try lstm cells as 16, it's amazing, the acc increased to more than 85% in 30 epochs.Of all the models I've tried before, only bidirectional lstm has achieved the highest accuracy of 83%.

Your advice is really amazing, I wonder to know  what theory your opinion about the cell of lstm you choose depends on, it's very useful and I also want to learn how to decide those hyperparameters,thank you very much!

**Comment by titu1994** [comment]
I am happy to hear that you were able to train a model to great accuracy in a relatively short amount of time.

Currently, we have not formed a formal method to decide if we should choose an ordinary LSTM cell verses an Attention LSTM cell. We encourage you to try both, but to know that the attention lstm has approximately 2 times the number of parameters as the regular LSTM, and therefore reduce the number of cells to prevent overfitting. 

We find that generally, self attention models help when there are long term dependencies, important 
information in only a subset of all the time steps or on highly noisy signals in the dataset.
